File: Travel Package Management Java/src/com/example/travel/PremiumPassenger.java
package com.example.travel;

public class PremiumPassenger extends Passenger {
    public PremiumPassenger(String name, int passengerNumber, double balance) {
        super(name, passengerNumber, balance);
    }

    @Override
    public boolean signUpForActivity(Destination destination, Activity activity) {
        if (!activity.isSignedUp(this)) {
            return activity.signUp(this);
        }
        return false;
    }

    @Override
    public double calculateActivityCost(Activity activity) {
        return 0.0; // Free for PremiumPassenger
    }
}

File: Travel Package management C++/PremiumPassenger.cpp
#include "PremiumPassenger.h"
#include "Activity.h"
#include "Destination.h"

using namespace std;


bool PremiumPassenger::signUpForActivity(shared_ptr<Destination> destination, shared_ptr<Activity> activity){
    if(!activity->isSignedUp(shared_from_this())){
        return activity->signUp(shared_from_this());
    }
    return false;
} 

double PremiumPassenger::calculateActivityCost(shared_ptr<Activity> activity){
    return 0.0;
}   
